pkgpanda/util.py

```diff
@@ -413,3 +414,74 @@ def hash_checkout(item):
         return hash_list(list(item))
     else:
         raise NotImplementedError("{} of type {}".format(item, type(item)))
+
+
+def split_by_token(token_prefix, token_suffix, string_, strip_token_decoration=False):
```

Comment:
I feel this is great improvement. 👍 

Some suggestions, and please see if you would like to consider them. (Also, feel free state a) if you agree, but it is not not worth it at this at this point, or b) it is a bad idea).


* Currently, defining token_prefix and token_suffix left to client. So, I can use the split_by_token like this.

```
split_by_token("foo", "bar", "foo something bar")
```
And I'd get the same string back.

* I am not sure if that flexibility is desirable. It can give way to negative test cases like above.

* Instead, if we define a well known tokens.

```
BLOCK_NAME_START = "{"
BLOCK_NAME_END = "}"

block_token = (BLOCK_NAME_START, BLOCK_NAME_END)

and split_by_token(block_token, "{something}")

```

We are restricting it to well-known-tokens that we define.  We can raise ValueError when a new kind of unknown token is encountered.